test/Microsoft.AspNetCore.Routing.Tests/LinkGeneratorTestBase.cs

```diff
@@ -77,16 +77,17 @@ private protected DefaultLinkGenerator CreateLinkGenerator(EndpointDataSource[]
             routeOptions = routeOptions ?? new RouteOptions();
             dataSources = dataSources ?? Array.Empty<EndpointDataSource>();
 
-            services.Configure<EndpointOptions>((o) =>
+            var options = Options.Create(routeOptions);
+            var serviceProvider = services.BuildServiceProvider();
+
+            if (dataSources != null || dataSources.Length > 0)
             {
-                for (var i = 0; i < dataSources.Length; i++)
+                var compositeEndpointDataSource = serviceProvider.GetRequiredService<CompositeEndpointDataSource>();
```

Comment:
I think there is an example of where there is value in an options type. To add a data source to the `CompositeEndpointDataSource` we need to fetch it from a service provider and then modify it. The optons `services.Configure<EndpointOptions>((o) => { ... })` syntax feels better.

Options could just expose the `IList<EndpointDataSource>` found on `CompositeEndpointDataSource`.